app/Http/Controllers/ClientController.php

```diff
@@ -58,7 +58,11 @@ public function index()
     public function getDatatable()
     {
         $search = Input::get('sSearch');
-        $userId = Auth::user()->filterId();
+
+        if(Auth::user()->hasPermission('view_client'))
```

Comment:
It may make sense to move this logic into the filterId function and pass the permission and entity type as parameters

Reply:
I implemented a filterIdByEntity function, this got missed, i'll need to patch this.